packages/gatsby-plugin-sharp/src/utils.js

```diff
@@ -26,3 +29,56 @@ export function createProgress(message, reporter) {
     },
   }
 }
+
+let progressBar
+let pendingImagesCounter = 0
+let firstPass = true
+const createOrGetProgressBar = reporter => {
+  if (!progressBar) {
+    progressBar = createGatsbyProgressOrFallbackToExternalProgressBar(
+      `Generating image thumbnails`,
+      reporter
+    )
+
+    const originalDoneFn = progressBar.done
+
+    // TODO this logic should be moved to the reporter.
+    // when done is called we remove the progressbar instance and reset all the things
+    // this will be called onPostBuild or when devserver is created
+    progressBar.done = () => {
+      originalDoneFn.call(progressBar)
+      progressBar = null
+      pendingImagesCounter = 0
+    }
+
+    progressBar.addImageToProcess = imageCount => {
```

Comment:
This is what we had in index file previously (moved it to utils because it's imported in few locations, so this avoids circular imports)

There is just addition of convenience `addImageToProcess` method to increase `total` (and start activity if needed) which would need to be copied to two places if it was not put here (or some other helper function)